snuba/query/processors/tagsmap.py

```diff
@@ -0,0 +1,125 @@
+import re
+
+from enum import Enum
+from typing import Optional, List, NamedTuple
+
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.datasets.transactions_processor import escape_field
+from snuba.query.types import Condition
+from snuba.request.request_settings import RequestSettings
+from snuba.util import is_condition, is_function
+
+
+TAG_PATTERN = re.compile(r"([a-zA-Z0-9_\.]+)\[([a-zA-Z0-9_\.]+)\]")
```

Comment:
Can this be consolidated with `NESTED_COL_EXPR_RE`?

https://github.com/getsentry/snuba/blob/657344be7cb89f95246e95854a765a0e873237ab/snuba/datasets/tags_column_processor.py#L16

Reply:
Not exactly. This one is supposed to be parametric to the column neame (it should work for any nested column) and just identify the structure, while the NESTED_COL_EXPR_RE is meant to identify tags and contexts so it needs to hardcode `tags` and `contexts`. I can ensure we define the content in the same way though.

Reply:
I guess I thought wherever `NESTED_COL_EXPR_RE.match` is used, it could share the same general pattern (like `TAG_PATTERN` defined here) and it could be conditional on what the first match group is.

Reply:
ok, done

Reply:
👍 Thanks!